security/pkg/platform/gcp_test.go

```diff
@@ -122,3 +124,173 @@ func TestGetDialOptions(t *testing.T) {
 		}
 	}
 }
+
+func TestGcpGetRequestMetadata(t *testing.T) {
+	testCases := map[string]struct {
+		token       string
+		expectedErr string
+		expected    map[string]string
+	}{
+		"Good Identity": {
+			token:       "token",
+			expectedErr: "",
+			expected: map[string]string{
+				httpAuthHeader: "Bearer token",
+			},
+		},
+	}
+
+	for id, c := range testCases {
+		jwt := jwtAccess{
+			token: c.token,
+		}
+
+		metadata, err := jwt.GetRequestMetadata(context.Background(), "")
+		if len(c.expectedErr) > 0 {
+			if err == nil {
+				t.Errorf("%s: Succeeded. Error expected: %v", id, err)
+			} else if err.Error() != c.expectedErr {
+				t.Errorf("%s: incorrect error message: %s VS %s",
+					id, err.Error(), c.expectedErr)
+			}
+			continue
+		} else if err != nil {
+			t.Fatalf("%s: Unexpected Error: %v", id, err)
```

Comment:
Any reason we use `Fatalf` instead of `Errorf`

Reply:
The original test used to be more complicated using metadata, I used Fatalf  to avoid seg fault
But since I used reflect.DeepEqual. Used Errorf instead of Fatalf.